### ISSUE TITLE
Support: Correlations table grows exponentially

### ISSUE THREAD

**Issue by adalberto-l** [MISP/MISP#8275]
### Support Questions

Hi Team,
in my work enviroment correlations table grows exponentially Referring. [#3142](https://github.com/MISP/MISP/issues/3142).
I don't want lose this data (no drop table or disable correlation), how can I resolve it?

Thanks
Adalberto 

### MISP version

2.4.151

### Operating System

Ubuntu 

### Operating System version

20.04

### PHP version

7.4.3 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Benni0** [comment]
Hi,
I am currently experiencing the same problem.
There are 10475601 attributes in the attributes table (4860.55 MB) and the correlations table needs 291521.55 MB.

Why do we need the correlations table exactly? Is it only for faster correlation view?
In my opinion a join of the attribute value in the attributes table could bring the same results, maybe a little bit slower.
But I think with such big correlation tables there also wouldn't be much speed up.

For additional information, I tried to pull the default misp feeds from abuse.ch, which have highly correlated attributes and these are the main reasons for the high amount of correlations.

**Comment by adalberto-l** [comment]
> Hi, I am currently experiencing the same problem. There are 10475601 attributes in the attributes table (4860.55 MB) and the correlations table needs 291521.55 MB.
> 
> Why do we need the correlations table exactly? Is it only for faster correlation view? In my opinion a join of the attribute value in the attributes table could bring the same results, maybe a little bit slower. But I think with such big correlation tables there also wouldn't be much speed up.
> 
> For additional information, I tried to pull the default misp feeds from abuse.ch, which have highly correlated attributes and these are the main reasons for the high amount of correlations.

Hi,
you are probably right by the correcting indexing you should to have a good performance but I prefer to preserve the default view.
I will make an alter table operation to resize column id of the table Correlations, have you ever thought of this?

**Comment by Benni0** [comment]
@adalberto-l the intention behind the idea was to get rid of the correlations table without loosing functionality. Due to a change of the sql queries to join the attribute values with the attribute table itself instead of the correlation table the functionality should be the same. But I'm not sure if this would work for all usecases. I haven't had time to dig into the code and try to figure out if this would eventually work.
I'm still looking for a solution for this problem.

What do you want to achieve by resize the id column of the correlations table?

**Comment by adalberto-l** [comment]
@Benni0 the column id of correlations table is arrived at the maximun value therefore is a problem when add events. I would like to maintain active this table without lose any functionality of MISP and modify query, view,....
Could you suggested me an alternative workaround?

**Comment by Benni0** [comment]
@adalberto-l the default datatype of the id column is int. You can try to change the datatype to bigint, but I haven't tested this. 
I have currently no problem with the primary keys, I only want to reduce the database size.

I'm sorry I can't be of more help.

**Comment by iglocska** [comment]
Correlations are not always 1:1 matches, so a simple lookup against the values is not as straight forward. A longer while ago I was working on a rework of the correlations that should make it a bit more sensible, but need to find the time to finish it. 

With that said, those numbers indicate simply bad data clustering, if a feed pulls in the same data over and over again it's either a bad configuration of a non MISP format feed or a disadvantageous setup from the MISP feed producer.

**Comment by Benni0** [comment]
Of course, these numbers indicate that many events are sharing the same attributes. However, to call that bad clustering goes a little bit to far, in my opinion, since the default MISP feeds from abuse.ch are also causing many correlations. When I came up with this idea, I only thought of 1:1 correlations. Loosing all correlations which are not 1:1 would indeed be a drawback. 

Nevertheless, I do think most correlations are 1:1 and to have these would be better than having no correlations at all. Taking this into consideration, I would implement an option to display these correlations by joining the attributes table if correlations are disabled on the MISP instance. What's your opinion regarding this feature? Do you think it has a chance to get merged?

**Comment by packet-rat** [comment]
## RFI - Correlations between Attributes in Objects in Same Event?

Is the 'bug' where Correlations are occurring between the same Attribute values in multiple Objects in the same Event fixed?   This was leading to exponential Correlations 'Explosions' in our Phishing Campaign custom object Events.

**Comment by packet-rat** [comment]
I'll answer my own question:  Evidently not yet addressed:  https://github.com/MISP/MISP/issues/7531.
I'll 'kick' this again in that ticket.  For this Ticket have a look to see if this is happening in your data sets - It can increase your Correlations tables exponentially.
